Src/MailMergeLib/HtmlBodyBuilder.cs

```diff
@@ -189,8 +192,7 @@ private void ReplaceImgSrcByCid()
 
```

Comment:
The following unit test does not throw here:

```CSharp
[Test]
public void EmbeddedDataImage_ShouldNotBeTouched()
{
	var image =
		"data:image/png;base64,\r\niVBORw0KGgoAAAANSUhEUgAAAAEAAAAKAQMAAABPHKYJAAAABGdBTUEAALGPC/xhBQAAAAZQTFRFB\r\nPl7AAAAx+zuaAAAAAd0SU1FB9MEFwMzHmwS680AAAALSURBVBjTY2DABAAAFAABQpvU+wAAAABJRU\r\n5ErkJggg==";
	var imageTag = $"<img width=\"10\" height=\"10\" alt=\"1Pixel\" src=\"{image}\">";
	var mmm = new MailMergeMessage("", "plain text",
		$"<html><body>{imageTag}</body></html>");
	var hbb = new HtmlBodyBuilder(mmm, (object) null);
	var html = hbb.GetBodyPart();
	Assert.That(html.ToString(), Does.Contain(imageTag));
}
```
How does a unit test case look like, that throws an `UriFormatException`?

Reply:
@PhnXnhP Could you already have a look at test under v5.8.0, or show when a `UriFormatException`throws?

Reply:
The error occurs when the size of the embedded image exceeds a certain value. I don't know the exact limit, but it should be around 50 kb.

Reply:
Yep, that's the cause (from [Uri.cs](https://referencesource.microsoft.com/system/net/System/URI.cs.html)):

[c_MaxUriBufferSize](https://referencesource.microsoft.com/system/R/b20fe821f8f3baba.html) = 0xFFF0;

// we don;t work with >= 64k Uris
            if (length >= [c_MaxUriBufferSize](https://referencesource.microsoft.com/system/net/System/URI.cs.html#b20fe821f8f3baba))
                return [ParsingError](https://referencesource.microsoft.com/system/net/System/UriEnumTypes.cs.html#237c8dc4861d2cfa).[SizeLimit](https://referencesource.microsoft.com/system/net/System/UriEnumTypes.cs.html#bb3f018386485b14);